src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsEventNotificationConfig.java

```diff
@@ -0,0 +1,154 @@
+    private static final String DEFAULT_HEX_COLOR = "#ff0500";
+    private static final String DEFAULT_CUSTOM_MESSAGE = "Graylog Teams Notification";
+    private static final long DEFAULT_BACKLOG_SIZE = 0;
+
+    static final String INVALID_BACKLOG_ERROR_MESSAGE = "Backlog size cannot be less than zero";
+    static final String INVALID_WEBHOOK_ERROR_MESSAGE = "Specified Webhook URL is not a valid URL";
+    static final String INVALID_TEAMS_URL_ERROR_MESSAGE = "Specified Webhook URL is not a valid Teams URL";
+    static final String WEB_HOOK_URL = "https://teams.webhook.office.com/services/xxxx/xxxxxxxxxxxxxxxxxxx";
+    static final String FIELD_WEBHOOK_URL = "webhook_url";
+    static final String TEAMS_CUSTOM_MESSAGE = "custom_message";
+    static final String TEAMS_ICON_URL = "icon_url";
+    static final String TEAMS_BACKLOG_SIZE = "backlog_size";
+    static final String TEAMS_COLOR = "color";
+
+    @JsonProperty(TEAMS_BACKLOG_SIZE)
+    public abstract long backlogSize();
+
+    @JsonProperty(TEAMS_COLOR)
+    @NotBlank
+    public abstract String color();
+
+    @JsonProperty(FIELD_WEBHOOK_URL)
+    @NotBlank
+    public abstract String webhookUrl();
+
+    @JsonProperty(TEAMS_CUSTOM_MESSAGE)
+    public abstract String customMessage();
+
+    @JsonProperty(TEAMS_ICON_URL)
+    @Nullable
+    public abstract String iconUrl();
+
+    @Override
+    @JsonIgnore
+    public JobTriggerData toJobTriggerData(EventDto dto) {
+        return EventNotificationExecutionJob.Data.builder().eventDto(dto).build();
+    }
+
+    public static Builder builder() {
+        return Builder.create();
+    }
+
+    @Override
+    @JsonIgnore
+    public ValidationResult validate() {
+        ValidationResult validation = new ValidationResult();
+
+        URI webhookUri;
+        try {
+            webhookUri = new URI(webhookUrl());
+            if (webhookUri.getHost().toLowerCase().contains("office")) {
+                if (!TEAMS_PATTERN.matcher(webhookUrl()).find()) {
+                    validation.addError(FIELD_WEBHOOK_URL, INVALID_TEAMS_URL_ERROR_MESSAGE);
+                }
+            }
+
+        } catch (Exception ex) {
+            validation.addError(FIELD_WEBHOOK_URL, INVALID_WEBHOOK_ERROR_MESSAGE);
+        }
+
+        if (backlogSize() < 0) {
+            validation.addError(TEAMS_BACKLOG_SIZE, INVALID_BACKLOG_ERROR_MESSAGE);
+        }
+
+        return validation;
+    }
+
+    @AutoValue.Builder
+    public static abstract class Builder implements EventNotificationConfig.Builder<Builder> {
+        @JsonCreator
+        public static Builder create() {
+
+            return new AutoValue_TeamsEventNotificationConfig.Builder()
+                    .type(TYPE_NAME)
+                    .color(DEFAULT_HEX_COLOR)
+                    .webhookUrl(WEB_HOOK_URL)
+                    .customMessage(DEFAULT_CUSTOM_MESSAGE)
+                    .backlogSize(DEFAULT_BACKLOG_SIZE);
+        }
```

Comment:
The webhook URL and custom message defaults do not appear to be shown in the UI for new notifications. Do you know why this seems to be happening? The color appears to show.

![image](https://user-images.githubusercontent.com/3423655/173437809-ff15ef80-924d-4089-b2ce-e00a3a0a890b.png)

Reply:
The default values for the UI are set in the frontend code.  See TeamsNotificationForm.tsx